datafusion/physical-expr/src/aggregate/array_agg_ordered.rs

```diff
@@ -323,20 +335,32 @@ impl Accumulator for OrderSensitiveArrayAggAccumulator {
 impl OrderSensitiveArrayAggAccumulator {
     fn evaluate_orderings(&self) -> Result<ScalarValue> {
```

Comment:
@mustafasrepo Have you considered changing `values: Vec<ScalarValue>` and `ordering_values: Vec<Vec<ScalarValue>>,` from vec to vecdeque for OrderSensitiveArrayAggAccumulator too? If we can have VecDequeue here, we can avoid `into()` and reuse evalute_orderings() for both Ordersensitive And NthValue. They do the similar things instead one is Vec<T>, another is VecDeque<T>.